manticore/ethereum/__init__.py

```diff
@@ -905,6 +927,10 @@ def _transaction(self, sort, caller, value=0, address=None, data=None, gaslimit=
             if '_pending_transaction' in state.context:
                 raise EthereumError("This is bad. It should not be a pending transaction")
 
+            # Choose an address here, because it will be dependent on the caller's nonce in this state
+            if address is None:
+                address = world.new_address(caller)
+
             # Migrate any expression to state specific constraint set
             caller, address, value, data = self._migrate_tx_expressions(state, caller, address, value, data)
```

Comment:
I just noticed that this line overwrites the variables `caller`, `address`, `value,`, and `data`. Won't that screw up the call to `self._migrate_tx_expressions` during the next iteration? This is code currently in `master`; I didn't write it. @feliam, should I change the variable names to preserve the originals for each iteration?

Reply:
OHhhhh yeeeees 
```caller_migrated, address_migrated, value_migrated, data_migrated = self._migrate_tx_expressions(state, caller, address, value, data)```